server/lib/transactions.js

```diff
@@ -0,0 +1,71 @@
+const constants = require('../constants');
+
+const expenseType = require('../constants/transactions').type.EXPENSE;
+
+module.exports = app => {
+
+  const errors = app.errors;
+  const models = app.set('models');
+
+  function createFromPaidExpense(payoutMethod, paymentMethod, paymentResponse, expense) {
+    if (paymentResponse) {
+      switch (paymentResponse.paymentExecStatus) {
+        case 'COMPLETED':
+          break;
+
+        case 'CREATED':
+          /*
+           * When we don't provide a preapprovalKey (paymentMethod.token) to payServices['paypal'](),
+           * it creates a payKey that we can use to redirect the user to PayPal.com to manually approve that payment
+           * TODO We should handle that case on the frontend
+           */
+          throw new errors.BadRequest(`Please approve this payment manually on ${paymentResponse.paymentApprovalUrl}`);
+
+        default:
+          throw new errors.ServerError(`controllers.expenses.pay: Unknown error while trying to create transaction for expense ${expense.id}`);
+      }
+    }
+
+    return models.Transaction.create({
+      netAmountInGroupCurrency: -expense.amount * 100,
+      ExpenseId: expense.id,
+      // TODO remove #postmigration, info redundant with joined tables?
+      type: expenseType,
+      amount: expense.amount,
+      currency: expense.currency,
```

Comment:
FYI, we probably want to keep `amount` and `currency` in the `Transactions` even after migration. It is duplicated but useful to have in one place as a record of the original transaction (especially while dealing with multiple currencies).